client/app/components/parameters.js

```diff
@@ -135,14 +135,44 @@ function ParametersDirective($location, $uibModal) {
           if (scope.changed) {
             scope.changed({});
           }
+          const params = {};
           scope.parameters.forEach((param) => {
-            if (param.value !== null || param.value !== '') {
-              $location.search(`p_${param.name}`, param.value);
-            }
+            extend(params, param.toUrlParams());
           });
+          $location.search(params);
         }, true);
       }
 
+      // Ant Date/Time pickers need a callback that will be called when value changes.
+      // We can supply `param.setValue` directly, but we also need to trigger digest
+      // cycle. To connect Angular and React, we will use this factory of wrappers:
+      // it will create callback, assign it to parameter by its name, and also
+      // invalidate cache (we cannot create a function and bind it to each params
+      // in template because it will lead to infinite digest loop).
+      let paramValueSetters = {};
+      scope.$watch('parameters.length', () => {
+        // Invalidate cache if parameter count changes; here we watch for array length
+        // which is just number - it will not cost us a lot
+        paramValueSetters = {};
+      });
+      scope.createParamValueSetter = (param) => {
+        const name = param.name;
+        const result = paramValueSetters[name] || ((value) => {
+          // Every time look for a parameter by its name - it allows not to invalidate
+          // cache every time user changes parameter list
+          param = find(scope.parameters, p => p.name === name);
+          if (param) {
+            param.setValue(value);
+            scope.$applyAsync();
+          } else {
+            // Delete cached item if parameter with such name does not exist anymore
+            delete paramValueSetters[name];
+          }
+        });
+        paramValueSetters[name] = result;
+        return result;
+      };
```

Comment:
😮 

How about we wrap the parameter input rendering in its own component (`<ParameterInput param=param>`), then we can have a "static" callback, that already has reference to the parameter and can call `$applyAsync` along with `setValue`?

Reply:
Good idea. Should it be React or Angular component? :smile:

Reply:
:-) Ideally React so we don't have to rewrite later on, but it's main purpose as glue for Angular, so can be Angular too...

Reply:
Let it be Angular component - currently it's more "angular" than "react": we still need "ngModel" there (ad I don't want to refactor it now), we will need "$scope" there and so on.

Reply:
@arikfr Done. Also noticed and fixed small bug